x-pack/plugin/sql/qa/server/src/main/java/org/elasticsearch/xpack/sql/qa/FieldExtractorTestCase.java

```diff
@@ -94,10 +94,14 @@ public void testKeywordField() throws IOException {
         createIndexWithFieldTypeAndProperties("keyword", fieldProps, explicitSourceSetting ? indexProps : null);
         index("{\"keyword_field\":\"" + keyword + "\"}");
 
-        Map<String, Object> expected = new HashMap<>();
-        expected.put("columns", Arrays.asList(columnInfo("plain", "keyword_field", "keyword", JDBCType.VARCHAR, Integer.MAX_VALUE)));
-        expected.put("rows", singletonList(singletonList(ignoreAbove ? null : keyword)));
-        assertResponse(expected, runSql("SELECT keyword_field FROM test"));
+        if (explicitSourceSetting == false || enableSource) {
```

Comment:
Maybe you can extract a method, where you pass the expected Map and the query.

Reply:
Makes sense. Will look into simplifying the code here.